src/snowflake/snowpark/dataframe.py

```diff
@@ -3730,6 +3732,7 @@ def with_column(
         Args:
             col_name: The name of the column to add or replace.
             col: The :class:`Column` or :class:`table_function.TableFunctionCall` with single column output to add or replace.
+            keep_order: If ``True``, the original order of the columns in the DataFrame is preserved when reaplcing a column.
```

Comment:
nit: I feel `keep_order` is a bit unclear, what about`replace_column_inplace` or `keep_column_order`?

Reply:
good point